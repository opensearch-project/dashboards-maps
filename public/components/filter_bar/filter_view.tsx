/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiBadge } from '@elastic/eui';
import { i18n } from '@osd/i18n';
import React, { FC } from 'react';

interface Props {
  isDisabled: boolean;
  valueLabel: string;
  onRemove: () => void;
  [propName: string]: any;
}

export const FilterView: FC<Props> = ({
  isDisabled,
  onRemove,
  onClick,
  valueLabel,
  ...rest
}: Props) => {
  let title = i18n.translate('maps.filter.filterBar.moreFilterActionsMessage', {
    defaultMessage: 'Filter: {valueLabel}. Select for more filter actions.',
    values: { valueLabel },
  });

  if (isDisabled) {
    title = `${i18n.translate('maps.filter.filterBar.disabledFilterPrefix', {
      defaultMessage: 'Disabled',
    })} ${title}`;
  }

  return (
    <EuiBadge
      title={title}
      color="hollow"
      iconType="cross"
      iconSide="right"
      closeButtonProps={{
        // Removing tab focus on close button because the same option can be obtained through the context menu
        // Also, we may want to add a `DEL` keyboard press functionality
        tabIndex: -1,
      }}
      iconOnClick={onRemove}
      iconOnClickAriaLabel={i18n.translate('maps.filter.filterBar.filterItemBadgeIconAriaLabel', {
        defaultMessage: 'Delete',
      })}
      onClick={onClick}
      onClickAriaLabel={i18n.translate('maps.filter.filterBar.filterItemBadgeAriaLabel', {
        defaultMessage: 'Filter actions',
      })}
      {...rest}
    >
      {valueLabel}
    </EuiBadge>
  );
};
