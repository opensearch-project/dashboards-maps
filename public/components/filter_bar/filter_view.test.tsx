/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { shallow } from 'enzyme';
import React from 'react';
import { FilterView } from './filter_view';

describe('test filter display', function () {
  it('renders filter display on disabled', () => {
    const mockOnClick = jest.fn();
    const mockOnRemove = jest.fn();
    const filterBar = shallow(
      <FilterView
        isDisabled={true}
        valueLabel={'mylabel'}
        onClick={mockOnClick}
        onRemove={mockOnRemove}
      />
    );
    expect(filterBar).toMatchSnapshot();
  });
  it('renders filter display on enabled', () => {
    const mockOnClick = jest.fn();
    const mockOnRemove = jest.fn();
    const filterView = shallow(
      <FilterView
        isDisabled={false}
        valueLabel={'mylabel'}
        onClick={mockOnClick}
        onRemove={mockOnRemove}
      />
    );
    expect(filterView).toMatchSnapshot();
  });
});
