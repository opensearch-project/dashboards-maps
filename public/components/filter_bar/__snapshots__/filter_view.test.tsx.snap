// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test filter display renders filter display on disabled 1`] = `
<EuiBadge
  closeButtonProps={
    Object {
      "tabIndex": -1,
    }
  }
  color="hollow"
  iconOnClick={[MockFunction]}
  iconOnClickAriaLabel="Delete"
  iconSide="right"
  iconType="cross"
  onClick={[MockFunction]}
  onClickAriaLabel="Filter actions"
  title="Disabled Filter: mylabel. Select for more filter actions."
>
  mylabel
</EuiBadge>
`;

exports[`test filter display renders filter display on enabled 1`] = `
<EuiBadge
  closeButtonProps={
    Object {
      "tabIndex": -1,
    }
  }
  color="hollow"
  iconOnClick={[MockFunction]}
  iconOnClickAriaLabel="Delete"
  iconSide="right"
  iconType="cross"
  onClick={[MockFunction]}
  onClickAriaLabel="Filter actions"
  title="Filter: mylabel. Select for more filter actions."
>
  mylabel
</EuiBadge>
`;
